pkg/commands/compute/language_go.go

```diff
@@ -24,7 +25,14 @@ import (
 // NOTE: In the 5.x CLI releases we persisted the default to the fastly.toml
 // We no longer do that. In 6.x we use the default and just inform the user.
 // This makes the experience less confusing as users didn't expect file changes.
-const GoDefaultBuildCommand = "tinygo build -target=wasi -gc=conservative -o bin/main.wasm ./"
+const TinyGoDefaultBuildCommand = "tinygo build -target=wasi -gc=conservative -o bin/main.wasm ./"
+
+// GoDefaultBuildCommand is a build command compiled into the CLI binary so it
+// can be used as a fallback for customer's who have an existing C@E project and
+// who don't have a `scripts.build` defined.
+//
+// NOTE: Only used if `toolchain = "go"` and if Go 1.21+ is installed.
+const GoDefaultBuildCommand = "env GOARCH=wasm GOOS=wasip1 go build -o bin/main.wasm ./"
```

Comment:
I am not confident `env` is going to work on Windows, so it's worth checking this if CI doesn't already.

Reply:
Yeah, I wasn't confident this would work on Windows but CI (which runs the full test suite) seems to think there were no issues 🤔 

In general would we just remove the `env` and just prefix the build script with GOARCH/GOOS? Not sure if exec.Command needs `env`.

Reply:
`env` is required because `exec.Command` doesn't run a shell.  We could set the environment outside of the command-line, but that obviously complicates things if we want the end user control over things.

Reply:
You can set the environment with https://pkg.go.dev/os/exec#Cmd.Environ , although that would require parsing the command line with https://pkg.go.dev/github.com/google/shlex or something to extract the environment variable settings.

Reply:
@dgryski I think it's fine to set the environment without shlex because we know what the _default_ values for those should be (i.e. they're already hard-coded into the binary `GOARCH=wasm GOOS=wasip1`). So I'll look at doing that on Wednesday when I'm back from PTO.

Once the user starts defining their own build script, then things are a little out of our hands but I imagine if someone wants to define their own build script, then they could just do an `export` directly in their shell as a workaround. I'm not sure in practice how many people are going to stray from the pre-defined default paths in the CLI (for either Go or TinyGo)